src/main/java/net/sf/jabref/gui/fieldeditors/FileListEditor.java

```diff
@@ -218,13 +218,18 @@ public void actionPerformed(ActionEvent actionEvent) {
         menu.add(moveToFileDir);
         moveToFileDir.addActionListener(new MoveFileAction(frame, entryEditor, this, true));
 
-        JMenuItem deleteFile = new JMenuItem(Localization.lang("Delete local file"));
+        JMenuItem deleteFile = new JMenuItem(Localization.lang("Permanently delete local file"));
         menu.add(deleteFile);
         deleteFile.addActionListener(e -> {
             int row = getSelectedRow();
+
             // no selection
-            if (row != -1) {
+            if (row == -1) {
+                return;
+            }
 
+            int userConfirm = JOptionPane.showConfirmDialog(null, Localization.lang("Are you sure you want to delete this file?"), Localization.lang("Warning"), JOptionPane.YES_NO_OPTION);
+            if (userConfirm == JOptionPane.YES_OPTION) {
```

Comment:
There are just a few more Strings, but it really helps to improve usability a  lot! Here is a more detailed explanation . http://ux.stackexchange.com/questions/756/what-are-some-alternatives-to-the-phrase-are-you-sure-you-want-to-xyz-in-confi

See also https://github.com/koppor/jabref/issues/149